In [1]:
import pandas as pd

In [2]:
train = pd.read_csv("train.csv.zip")

In [3]:
train

ID_code  target    var_0   var_1    var_2   var_3    var_4  \
0            train_0       0   8.9255 -6.7863  11.9081  5.0930  11.4607   
1            train_1       0  11.5006 -4.1473  13.8588  5.3890  12.3622   
2            train_2       0   8.6093 -2.7457  12.0805  7.8928  10.5825   
3            train_3       0  11.0604 -2.1518   8.9522  7.1957  12.5846   
4            train_4       0   9.8369 -1.4834  12.8746  6.6375  12.2772   
...              ...     ...      ...     ...      ...     ...      ...   
199995  train_199995       0  11.4880 -0.4956   8.2622  3.5142  10.3404   
199996  train_199996       0   4.9149 -2.4484  16.7052  6.6345   8.3096   
199997  train_199997       0  11.2232 -5.0518  10.5127  5.6456   9.3410   
199998  train_199998       0   9.7148 -8.6098  13.6104  5.7930  12.5173   
199999  train_199999       0  10.8762 -5.7105  12.1183  8.0328  11.5577   

          var_5   var_6    var_7  ...  var_190  var_191  var_192  var_193  \
0       -9.2834  5.1187  18.6266  ...   4.4354   3.9642   3.1364   1.6910   
1        7.0433  5.6208  16.5338  ...   7.6421   7.7214   2.5837  10.9516   
2       -9.0837  6.9427  14.6155  ...   2.9057   9.7905   1.6704   1.6858   
3       -1.8361  5.8428  14.9250  ...   4.4666   4.7433   0.7178   1.4214   
4        2.4486  5.9405  19.2514  ...  -1.4905   9.5214  -0.1508   9.1942   
...         ...     ...      ...  ...      ...      ...      ...      ...   
199995  11.6081  5.6709  15.1516  ...   6.1415  13.2305   3.9901   0.9388   
199996 -10.5628  5.8802  21.5940  ...   4.9611   4.6549   0.6998   1.8341   
199997  -5.4086  4.5555  21.5571  ...   4.0651   5.4414   3.1032   4.8793   
199998   0.5339  6.0479  17.0152  ...   2.6840   8.6587   2.7337  11.1178   
199999   0.3488  5.2839  15.2058  ...   8.9842   1.6893   0.1276   0.3766   

        var_194  var_195  var_196  var_197  var_198  var_199  
0       18.5227  -2.3978   7.8784   8.5635  12.7803  -1.0914  
1       15.4305   2.0339   8.1267   8.7889  18.3560   1.9518  
2       21.6042   3.1417  -6.5213   8.2675  14.7222   0.3965  
3       23.0347  -1.2706  -2.9275  10.2922  17.9697  -8.9996  
4       13.2876  -1.5121   3.9267   9.5031  17.9974  -8.8104  
...         ...      ...      ...      ...      ...      ...  
199995  18.0249  -1.7939   2.1661   8.5326  16.6660 -17.8661  
199996  22.2717   1.7337  -2.1651   6.7419  15.9054   0.3388  
199997  23.5311  -1.5736   1.2832   8.7155  13.8329   4.1995  
199998  20.4158  -0.0786   6.7980  10.0342  15.5289 -13.9001  
199999  15.2101  -2.4907  -2.2342   8.1857  12.1284   0.1385  

[200000 rows x 202 columns]

In [4]:
train.columns

Index(['ID_code', 'target', 'var_0', 'var_1', 'var_2', 'var_3', 'var_4',
       'var_5', 'var_6', 'var_7',
       ...
       'var_190', 'var_191', 'var_192', 'var_193', 'var_194', 'var_195',
       'var_196', 'var_197', 'var_198', 'var_199'],
      dtype='object', length=202)

In [5]:
train.iloc[:,0:5]

ID_code  target    var_0   var_1    var_2
0            train_0       0   8.9255 -6.7863  11.9081
1            train_1       0  11.5006 -4.1473  13.8588
2            train_2       0   8.6093 -2.7457  12.0805
3            train_3       0  11.0604 -2.1518   8.9522
4            train_4       0   9.8369 -1.4834  12.8746
...              ...     ...      ...     ...      ...
199995  train_199995       0  11.4880 -0.4956   8.2622
199996  train_199996       0   4.9149 -2.4484  16.7052
199997  train_199997       0  11.2232 -5.0518  10.5127
199998  train_199998       0   9.7148 -8.6098  13.6104
199999  train_199999       0  10.8762 -5.7105  12.1183

[200000 rows x 5 columns]

# Problems

1.          \begin{equation}
            P(target = 0 | (1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18))   
            \end{equation}
            
            
            

2.          \begin{equation}
            P((1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18) | target = 0)
            \end{equation}
            
            
            

3.          \begin{equation}
            P(target = 1 | (1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18))
            \end{equation}
            
            
            
            
4.          \begin{equation}
            P((1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18) | target = 1)
            \end{equation}

# Using Conditional Probability
\begin{equation}
P(A|B) = \frac{P(A \cap B)}{P(B)}
\end{equation}

# For question 1.

\begin{equation}
P(A \cap B) = P((target = 0)\cap (1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18))
\end{equation}

In [6]:
P1_1 = (train['target'] == 0) &  (train['var_0'] >= 1) & (train['var_0'] <= 15)  & (train['var_1'] >= -10) & (train['var_1'] <= 9.5) & (train['var_2'] >= 4) & (train['var_2'] <= 18)

In [7]:
A_1 = train[P1_1].shape[0]/train.shape[0]

In [8]:
A_1

0.819455

\begin{equation}
P(A \cap B)=0.819455
\end{equation}

\begin{equation}
P(B) = P((1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18))
\end{equation}

In [9]:
P1_2 = (train['var_0']>= 1) & (train['var_0']<= 15)  & (train['var_1']>= -10) & (train['var_1'] <= 9.5) & (train['var_2'] >= 4) & (train['var_2'] <= 18)

In [10]:
B_1 = train[P1_2].shape[0]/train.shape[0]

In [11]:
B_1

0.905805

\begin{equation}
P(B)=0.905805
\end{equation}

In [12]:
A_1/B_1

0.9046704312738394

\begin{equation}
P(A|B)..1 = 0.9046704312738394
\end{equation}

# For question 2.

\begin{equation}
P(A \cap B) = P((1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18) \cap (target = 0))
\end{equation}


In [13]:
P2_1 = (train['var_0'] >= 1) & (train['var_0'] <= 15)  & (train['var_1'] >= -10) & (train['var_1'] <= 9.5) & (train['var_2'] >= 4) & (train['var_2'] <= 18) & (train['target'] == 0) 

In [14]:
A_2 = train[P2_1].shape[0]/train.shape[0]

In [15]:
A_2

0.819455

\begin{equation}
P(A \cap B) = 0.819455
\end{equation}

\begin{equation}
P(B) =  (target = 0)
\end{equation}

In [16]:
P2_2 = train['target']== 0

In [17]:
B_2 = train[P2_2].shape[0]/train.shape[0]

In [18]:
B_2

0.89951

\begin{equation}
P(B) = 0.89951
\end{equation}

In [19]:
A_2/B_2

0.9110015452857667

\begin{equation}
P(A|B)..2 = 0.9110015452857667
\end{equation}

# For question 3.
\begin{equation}
P(A \cap B) = P((target = 1) \cap (1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18))
\end{equation}

In [20]:
P3_1 = (train['target'] == 1) &  (train['var_0'] >= 1) & (train['var_0'] <= 15)  & (train['var_1'] >= -10) & (train['var_1'] <= 9.5) & (train['var_2'] >= 4) & (train['var_2'] <= 18)

In [21]:
A_3 = train[P3_1].shape[0]/train.shape[0]

In [22]:
A_3

0.08635

\begin{equation}
P(A \cap B) = 0.08635
\end{equation}

\begin{equation}
P(B) = P((1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18))
\end{equation}

In [23]:
P3_2 = (train['var_0']>= 1) & (train['var_0']<= 15)  & (train['var_1']>= -10) & (train['var_1'] <= 9.5) & (train['var_2'] >= 4) & (train['var_2'] <= 18)

In [24]:
B_3 = train[P3_2].shape[0]/train.shape[0]

In [25]:
B_3

0.905805

\begin{equation}
P(B)=0.905805
\end{equation}

In [26]:
A_3/B_3

0.0953295687261607

\begin{equation}
P(A|B)..3 = 0.0953295687261607
\end{equation}

# For question 4.
\begin{equation}
P(A \cap B) = P((1 \leq var_0 \leq 15) \cap (-10 \leq var_1 \leq 9.5) \cap (4 \leq var_2 \leq 18) \cap (target = 1))
\end{equation}

In [27]:
P4_1 = (train['var_0'] >= 1) & (train['var_0'] <= 15)  & (train['var_1'] >= -10) & (train['var_1'] <= 9.5) & (train['var_2'] >= 4) & (train['var_2'] <= 18) & (train['target'] == 1) 

In [28]:
A_4 = train[P4_1].shape[0]/train.shape[0]

In [29]:
A_4

0.08635

\begin{equation}
P(A \cap B) = 0.08635
\end{equation}

\begin{equation}
P(B) =  (target = 1)
\end{equation}

In [30]:
P4_2 = train['target']== 1

In [31]:
B_4 = train[P4_2].shape[0]/train.shape[0]

In [32]:
B_4

0.10049

\begin{equation}
P(B) = 0.10049
\end{equation}

In [33]:
A_4/B_4

0.8592894815404518

\begin{equation}
P(A|B)..4 = 0.8592894815404518
\end{equation}